src/coreclr/classlibnative/float/floatdouble.cpp

```diff
@@ -280,6 +280,24 @@ FCIMPL1_V(double, COMDouble::Sin, double x)
     return (double)sin(x);
 FCIMPLEND
 
+/*====================================SinCos====================================
+**
+==============================================================================*/
+FCIMPL1_V(COMDouble::SinCosResult, COMDouble::SinCos, double x)
+    FCALL_CONTRACT;
+
+    SinCosResult result;
+
+#ifdef _MSC_VER
+    result.Sin = (double)sin(x);
+    result.Cos = (double)cos(x);
+#else
+    sincos(x, &result.Sin, &result.Cos);
```

Comment:
```suggestion
#elif __APPLE__
    __sincos(x, &result.Sin, &result.Cos);
#else
    sincos(x, &result.Sin, &result.Cos);
```

Reply:
Unfortunately, it isn't this simple. Stuff on Unix has to go through the PAL layer so I need effectively something like the following in PAL instead:
```cpp
#if defined(__APPLE__)
#define sincos __sincos
#endif
```

Reply:
Ah right: https://github.com/dotnet/runtime/issues/43169#issuecomment-705761640